lib/experimental/new_embed.dart

```diff
@@ -628,21 +649,32 @@ class NewEmbed {
 
     if (focus) userCodeEditor.focus();
   }
+
+  void _initializeMaterialRipples() {
+    MDCRipple(executeButton._element.element);
+    MDCRipple(reloadGistButton._element.element);
+    MDCRipple(formatButton._element.element);
+    MDCRipple(showHintButton._element.element);
+  }
 }
 
 // material-components-web uses specific classes for its navigation styling,
 // rather than an attribute. This class extends the tab controller code to also
 // toggle that class.
 class NewEmbedTabController extends TabController {
+  final MDCTabBar _tabBar;
+
+  NewEmbedTabController(this._tabBar);
+
   /// This method will throw if the tabName is not the name of a current tab.
   @override
   void selectTab(String tabName) {
-    TabElement tab = tabs.firstWhere((t) => t.name == tabName);
+    var tab = tabs.firstWhere((t) => t.name == tabName);
```

Comment:
nit: 672-679 effectively loop over the tabs twice (once looking for a specific tab, once to do the `aria-selected` assignment. Consider collapsing this section into a single for-loop for readability:

```dart
for (int idx = 0; idx < tabs.length; idx++) {
  if (tabs[idx].name == tabName) {
    _tabBar.activateTab(idx);
    tabs[idx].toggleAttr('aria-selected', true);
  } else {
    tabs[idx].toggleAttr('aria-selected', false);
  }
}
```

Reply:
that's true, but I think this is more readable and doesn't impact performance. (unless we are going to have a *lot* of tabs :) )

Reply:
Fair enough. For something like this, it's author discretion.